### PR TITLE
chore: add pending geofence metric file-backed queue

### PR DESCRIPTION
**Pull request by mrehan27**
## Stack
Part of the MBL-1628 four-PR stack. Each PR layers one concern; merge in order:

- [#1060](https://github.com/customerio/customerio-ios/pull/1060) — geofence transition event delivery via EventBus (base for this PR, in review)
- **This PR** — file-backed pending-metric queue (persistence primitive)
- _(next)_ — direct-HTTP `GeofenceDeliveryTracker` (uses the queue's model)
- _(next)_ — wire `GeofenceEventTracker` to the three-layer flow + flush hook on init

## Ticket
[MBL-1628 — Send Geofence Transition Events](https://linear.app/customerio/issue/MBL-1628/ios-send-geofence-transition-events)

## Summary
Adds the file-backed queue layer of the three-layer geofence transition delivery design: a `PendingGeofenceMetric` model and a `PendingGeofenceMetricStore` actor that persists transitions before any HTTP attempt and retains them until the direct-HTTP layer (next PR) confirms success.

This PR adds only the persistence primitive — no wiring yet, no existing code path references it.

## Design rationale
- **File over `UserDefaults`** — written with `completeUntilFirstUserAuthentication` Data Protection so geofence callbacks that fire before the first unlock can still read after, and excluded from iCloud/iTunes backups (transient device-local context, not user data).
- **Single actor, no internal `await`** — every public method does load → modify → save in one actor-isolated step, so concurrent callers can't observe a half-applied state or lose a write. Same pattern as `GeofenceStorage`.
- **100-entry FIFO cap, drops oldest on overflow** — bounds disk usage if the device stays offline for a long time.
- **`.secondsSince1970` date strategy** — matches `GeofenceStorage` for cross-store consistency.

## Scope
Two new source files; nothing else touched. Not yet referenced by any module init or other code path — the wire-up PR adds the only call site.

## Diff size
- Source: ~180 lines (1 model + 1 actor)
- Tests: ~260 lines, excluded from the 250-line cap

## Test plan
- [x] `make generate && make format && make lint` clean
- [x] Targeted: `LocationTests/PendingGeofenceMetricStoreTests` — 13/13 pass (~1s)
- [x] Coverage: append/load/remove/clearAll happy paths, capacity boundary at exactly N (off-by-one guard), capacity overflow drops oldest, persistence across actor instances, concurrent ops stress test asserts capacity invariant holds

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> Low risk because this PR only adds new, currently-unwired persistence code and tests; the main risk is potential file I/O/serialization edge cases when it gets integrated later.
> 
> **Overview**
> Adds a new persistence primitive for geofence transition delivery: a `PendingGeofenceMetric` model plus an actor-backed `PendingGeofenceMetricStore` that queues metrics in a JSON file under Application Support with a 100-item FIFO cap.
> 
> The store supports append/load/remove/clear operations, uses `.secondsSince1970` date encoding, applies `completeUntilFirstUserAuthentication` file protection, and excludes the directory/file from backups; comprehensive unit tests cover capacity, persistence across instances, and concurrent access invariants.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 08fecb6656b17b88181aee2fb4d8ae6875a28718. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: Sources/Location/Geofence/Storage/PendingGeofenceMetric.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import CioInternalCommon
+import Foundation
+
+/// A geofence transition queued for direct-HTTP delivery.
+struct PendingGeofenceMetric: Codable, Equatable, Sendable {
+    let id: UUID
```
Comment (mahmoud-elmorabea) [inline Sources/Location/Geofence/Storage/PendingGeofenceMetric.swift:6]: Do we need that extra id? Should the `geofenceId` be enough?

**Review by mahmoud-elmorabea — APPROVED** [review]

File: Sources/Location/Geofence/Storage/PendingGeofenceMetric.swift (hunk 1 above)
Comment (mrehan27) [reply Sources/Location/Geofence/Storage/PendingGeofenceMetric.swift:6]: You're right that the extra UUID isn't necessary — Android uses a composite key (`geofenceId_transition_timestamp_ms`) that serves the same per-row identity and adds a second line of defense at the storage layer if cooldown ever slips. Will align iOS in the cleanup PR.
